src/Fantomas.Core/CodePrinter.fs

```diff
@@ -73,6 +73,8 @@ let genTrivia (trivia: TriviaNode) (ctx: Context) =
 
     // Some items like #if or Newline should be printed on a newline
     // It is hard to always get this right in CodePrinter, so we detect it based on the current code.
+    // However this may cause duplicate newlines in case of content-after trivia of a node
```

Comment:
This isn't entirely true and I find these extra lines to be rather confusing.
Please remove them.